javascript/modules/client/src/client-erc20.ts

```diff
@@ -195,14 +201,27 @@ export class ClientErc20 extends ClientCore implements IClientErc20 {
     }
 
     const mockAddresses = [
-      "0x8367dc645e31321CeF3EeD91a10a5b7077e21f70",
+      "0x1234...",
       "0xDA9dfA130Df4dE4673b89022EE50ff26f6EA73Cf",
       "0xBE0eB53F46cd790Cd13851d5EFf43D12404d33E8",
       "0x2dB75d8404144CD5918815A44B8ac3f4DB2a7FAf",
       "0xc1d60f584879f024299DA0F19Cdb47B931E35b53",
     ];
 
-    return Promise.resolve(mockAddresses.filter(() => Math.random() > 0.4));
+    return new Promise(resolve => setTimeout(resolve, 1000)).then(() =>
+      mockAddresses.filter(() => Math.random() > 0.4)
+    );
+  }
+
+  private _getProposals(daoAddressOrEns: string) {
+    if (!daoAddressOrEns) {
+      throw new Error("Invalid DAO address or ENS");
+    }
+
+    setProposalStatus(MOCK_PROPOSALS);
```

Comment:
Modifying an object passed by reference is a bit of an anti pattern. Better to have a new variable containing the result, which is a copy of the original. 

At this stage it doesn't matter, but in general, modifying a global object can have unintended side effects elsewhere